### PR TITLE
GPTQ example refinements

### PR DESCRIPTION
**Pull request by YIYANGCAI**
## Type of Change

bug fix & enhancements

## Description

1. move gptq dataset alignments process in gptq.py
2. fix some bugs to fit more models (mpt, etc.)
3. add "block_size" as a user-available parameter
4. modify "device" as a user-available parameter, since GPTQ model's most part will be always on CPU.
5. update requirements, since running some models (llama-2-7b, falcon-7b, etc.) requires packages with more advanced version.

## Expected Behavior & Potential Risk

GPTQ can be called by itrex in better manner.

## How has this PR been tested?

PreCI and UTs

## Dependency Change?

None

### CODE REVIEW

**Comment by YIYANGCAI** [comment]
run-gptq-llm.sh under root directory is for convenience and will be removed when this pr is ready.

**Comment by YIYANGCAI** [comment]
Related enhancements is linked with this [pr98](https://github.com/intel/intel-extension-for-transformers/pull/98) in itrex

**Review by xin3he — APPROVED** [review]

File: run-gptq-llm.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export CUDA_VISIBLE_DEVICES=6
```
Comment (wenhuach21) [inline run-gptq-llm.sh:1]: 6->0 or remove

**Review by xin3he — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/gptq.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 from tqdm import tqdm
 from functools import partial
 from ...utils import logger
+import random
 
 DEBUG = False
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/gptq.py:31]: better call RTN in weight_only.py instead

File: neural_compressor/adaptor/torch_utils/gptq.py [hunk 3]
```diff
@@ -260,9 +359,17 @@ def forward(layer, hidden_states, **kwargs):
             self.cache['i'] += 1
             for arg in kwargs:
                 # TODO: investigate include parameters
-                if isinstance(kwargs[arg], torch.Tensor) or arg == "alibi":
-                    self.cache[arg] = kwargs[arg]
+                if self.use_full_length:
+                    if isinstance(kwargs[arg], torch.Tensor) or arg == "alibi":
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/gptq.py:363]: is this enough, shall we cache all the kwargs?

**Review by wenhuach21 — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/gptq.py (hunk 2 above)
Comment (wenhuach21) [reply neural_compressor/adaptor/torch_utils/gptq.py:31]: BTW, it seems that we haven't supported "true sequential" option

File: docs/source/quantization_weight_only.md [hunk 4]
```diff
@@ -76,9 +85,8 @@ conf = PostTrainingQuantConfig(
             },
         },
     },
-    ### GPTQ is WIP
     recipes={
-        # 'gptq_args':{'percdamp': 0.01},
+        # 'gptq_args':{'percdamp': 0.01, 'actorder':True, 'block_size': 128, 'nsamples': 128, 'use_full_length': False},
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:89]: better to change "use full length " to "use max length"?

File: docs/source/quantization_weight_only.md [hunk 5]
```diff
@@ -47,8 +47,17 @@ There are many excellent works for weight only quantization to improve its accur
 |  mse_range |      True     | Whether search for the best clip range from range [0.89, 1.0, 0.01] |
 |  n_blocks  |       5       |   Split the model into n blocks for AWQ search to avoid out-of-memory   |
 
+**GPTQ arguments**:
+|  gptq_args  | default value |                               comments                              |
+|:----------:|:-------------:|:-------------------------------------------------------------------:|
+| actorder | False |   Whether to sort Hessian's diagonal values to rearrange channel-wise quantization order|
+|  percdamp | 0.01 | Add a small Hessian's diagonal values to itself, which increase Hessian's numerical stability|
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:54]: what does percdamp mean? damp?

File: docs/source/quantization_weight_only.md [hunk 6]
```diff
@@ -47,8 +47,17 @@ There are many excellent works for weight only quantization to improve its accur
 |  mse_range |      True     | Whether search for the best clip range from range [0.89, 1.0, 0.01] |
 |  n_blocks  |       5       |   Split the model into n blocks for AWQ search to avoid out-of-memory   |
 
+**GPTQ arguments**:
+|  gptq_args  | default value |                               comments                              |
+|:----------:|:-------------:|:-------------------------------------------------------------------:|
+| actorder | False |   Whether to sort Hessian's diagonal values to rearrange channel-wise quantization order|
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:53]: better to use act_order? the naming of parameters can be more unified.

Comment (YIYANGCAI) [reply docs/source/quantization_weight_only.md:53]: resolved

File: docs/source/quantization_weight_only.md (hunk 5 above)
Comment (YIYANGCAI) [reply docs/source/quantization_weight_only.md]: added more explanation in quantization_weight_only.md

File: docs/source/quantization_weight_only.md (hunk 4 above)
Comment (YIYANGCAI) [reply docs/source/quantization_weight_only.md:90]: resolved

**Comment by YIYANGCAI** [comment]
Several things can be solved by subsequential prs: 1. use INC-RTN to calculate GPTQ's block-wise quantization parameters. 2. consider add true-sequential function. 3. refine kwargs related input collection.

**Pull request merged**